tests/basic/storage_pool_test.go

```diff
@@ -8828,3 +8828,141 @@ var _ = Describe("{ReplResyncOnPoolExpand}", func() {
+		}()
+		err = Inst().S.AddLabelOnNode(*selectedNode, k8s.NodeType, k8s.FastpathNodeType)
+		log.FailOnError(err, fmt.Sprintf("Failed add label on node %s", selectedNode.Name))
+		err = Inst().S.AddLabelOnNode(secondReplNode, k8s.NodeType, k8s.FastpathNodeType)
+		log.FailOnError(err, fmt.Sprintf("Failed add label on node %s", secondReplNode.Name))
+
+		isjournal, err := isJournalEnabled()
+		log.FailOnError(err, "is journal enabled check failed")
+
+		err = adjustReplPools(*selectedNode, secondReplNode, isjournal)
+		log.FailOnError(err, "Error setting pools for clean volumes")
+
+		Inst().AppList = []string{"fio-fastpath"}
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("sfullad-%d", i))...)
+		}
+		defer appsValidateAndDestroy(contexts)
+
+		err = WaitForPoolOffline(*selectedNode)
+		log.FailOnError(err, fmt.Sprintf("Failed to make node %s storage down", selectedNode.Name))
+
+		poolsStatus, err := Inst().V.GetNodePoolsStatus(*selectedNode)
+		log.FailOnError(err, "error getting pool status on node %s", selectedNode.Name)
+
+		var offlinePoolUUID string
+		for i, s := range poolsStatus {
+			if s == "Offline" {
+				offlinePoolUUID = i
+				break
+			}
+		}
+		selectedPool, err := GetStoragePoolByUUID(offlinePoolUUID)
+		log.FailOnError(err, "error getting pool with UUID [%s]", offlinePoolUUID)
+
+		defer func() {
+			status, err := Inst().V.GetNodePoolsStatus(*selectedNode)
+			log.FailOnError(err, fmt.Sprintf("error getting node %s pool status", selectedNode.Name))
+			log.InfoD(fmt.Sprintf("Pool %s has status %s", selectedNode.Name, status[selectedPool.Uuid]))
+			if status[selectedPool.Uuid] == "In Maintenance" {
+				log.InfoD(fmt.Sprintf("Exiting pool maintenance mode on node %s", selectedNode.Name))
+				err = Inst().V.ExitPoolMaintenance(*selectedNode)
+				log.FailOnError(err, fmt.Sprintf("fail to exit pool maintenance mode ib node %s", selectedNode.Name))
+			}
+		}()
+
+		log.InfoD(fmt.Sprintf("Entering pool maintenance mode on node %s", selectedNode.Name))
+		err = Inst().V.EnterPoolMaintenance(*selectedNode)
+		log.FailOnError(err, fmt.Sprintf("fail to enter node %s in maintenance mode", selectedNode.Name))
+		status, err := Inst().V.GetNodePoolsStatus(*selectedNode)
+		log.FailOnError(err, fmt.Sprintf("error getting node %s pool status", selectedNode.Name))
+		log.InfoD(fmt.Sprintf("pool %s status %s", selectedNode.Name, status[selectedPool.Uuid]))
+
+		stepLog = fmt.Sprintf("expand pool %s using add-disk", selectedPool.Uuid)
+		var expandedExpectedPoolSize uint64
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			expandedExpectedPoolSize = (selectedPool.TotalSize / units.GiB) * 2
+
+			log.FailOnError(err, "Failed to check if Journal enabled")
+
+			log.InfoD("Initiate add cloud drive and validate")
+			err := addCloudDrive(*selectedNode, selectedPool.ID)
```

Comment:
make the changes to create a new pool rather than adding disk to existing pool.
since legacy add drive to existing pool is not the recommended way